### PR TITLE
Editors Label - Fix editor doesn't provide accessibility information about it's label

### PR DESCRIPTION
**Pull request by groshenkovamarina**

### CODE REVIEW

File: js/ui/text_box/ui.text_editor.base.js [hunk 1]
```diff
@@ -447,7 +448,7 @@ const TextEditorBase = Editor.inherit({
 
 
     _cleanLabelObservable: function() {
```
Comment (ksercs) [inline js/ui/text_box/ui.text_editor.base.js]:
```suggestion
    _unobserveLabelContainerResize: function() {
```

File: js/ui/text_box/ui.text_editor.base.js [hunk 2]
```diff
@@ -447,7 +448,7 @@ const TextEditorBase = Editor.inherit({
...
-        if(this._labelContainerElement) {
+        if(this._labelContainerElement && this._label.isVisible()) {
```
Comment (ksercs) [inline js/ui/text_box/ui.text_editor.base.js]: Can it really be no label container element?

Comment (ksercs) [inline js/ui/text_box/ui.text_editor.base.js]: This code does not unobserve label container if label became invisible in runtime.

File: js/ui/text_box/ui.text_editor.base.js [hunk 3]
```diff
@@ -492,7 +500,9 @@ const TextEditorBase = Editor.inherit({
 
         this._label = new TextEditorLabelCreator(labelConfig);
 
-        if(this._labelContainerElement) {
+        if(this._label.isVisible()) {
+            this._setLabelContainerAria(labelId);
+
             resizeObserverSingleton.observe(this._labelContainerElement, this._updateLabelWidth.bind(this));
```
Comment (ksercs) [inline js/ui/text_box/ui.text_editor.base.js]: What about label visibility runtime change?

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js [hunk 4]
```diff
@@ -1378,16 +1380,18 @@ QUnit.module('Lookup', {
 
 QUnit.module('label integration', () => {
     QUnit.test('lookup should pass containerWidth equal to field width', function(assert) {
-        const constructorMock = sinon.stub();
-        Lookup.mockTextEditorLabel(constructorMock);
+        this.TextEditorLabelMock = (args) => { this.labelArgs = args; return new TextEditorLabel(args); };
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js]: seems like we can write smth like 
```suggestion
        this.TextEditorLabelMock = sinon.spy((args) =>  new TextEditorLabel(args));
```

and get rid of `this.labelArgs`

File: testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 5]
```diff
@@ -477,17 +478,31 @@ QUnit.module('label integration', {
...
+            updateMark = sinon.stub()
+            updateContainsButtonsBefore = sinon.stub()
+        }
+
+        this.TextEditorLabelMock = (args) => { this.labelArgs = args; return this.labelMock = new TextEditorLabelMock(args); };
+
+        TextEditor.mockTextEditorLabel(this.TextEditorLabelMock);
+    },
+    afterEach: function() {
+        Object.values(this.labelMock, (stub) => {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]: We can create a sinon [SandBox](https://sinonjs.org/releases/latest/sandbox/) in beforeEach to simplify the clearing

File: testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 6]
```diff
@@ -531,36 +547,44 @@ QUnit.module('label integration', {
...
-        },
-        afterEach: function() {
-            Object.values(this.LabelMock, (stub) => {
-                stub.reset();
-            });
-            TextEditor.restoreTextEditorLabel();
         }
     }, () => {
         QUnit.test('width', function(assert) {
+            this.init();
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js]: we already called `init` on beforeEach

File: js/ui/text_box/ui.text_editor.base.js (hunk 2 above)
Comment (groshenkovamarina) [reply js/ui/text_box/ui.text_editor.base.js]: Yes, at the first time we call _cleanLabelObservable early than we define this._labelContainerElement

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js (hunk 4 above)
Comment (groshenkovamarina) [reply testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js]: in this test we check label args

**Pull request closed without merging**